File: frontend/libs/portal/model/Checklist.ts
// SPDX-FileCopyrightText: 2025 Mercedes-Benz Group AG and Mercedes-Benz AG
//
// SPDX-License-Identifier: Apache-2.0

export class Checklist {
  _key = '';
  updated?: string;
  created?: string;
  name = '';
  nameDE = '';
  description = '';
  descriptionDE = '';
  policyLabels: string[] = [];
  items: ChecklistItem[] = [];
  active: boolean = false;
}

export class ChecklistItem {
  _key = '';
  updated?: string;
  created?: string;
  name = '';
  triggerType = '';
  policyStatus: string[] = [];
  policyLabels: string[] = [];
  classifications: string[] = [];
  scanRemarks = '';
  licenseIds: string[] = [];
  componentNames: string[] = [];
  targetTemplateName = '';
  targetTemplateKey = '';
}

export enum TriggerTypes {
  DEFAULT = 'DEFAULT',
  CLASS_AND = 'CLASS_AND',
  CLASS_OR = 'CLASS_OR',
  POLICY_STATUS = 'POLICY_STATUS',
  SCAN_REMARK = 'SCAN_REMARK',
  LICENSE = 'LICENSE',
  COMPONENT_NAME = 'COMPONENT_NAME',
}

export enum PolicyStatusTypes {
  ALLOWED = 'ALLOWED',
  DENIED = 'DENIED',
  UNASSERTED = 'UNASSERTED',
  WARNED = 'WARNED',
  QUESTIONED = 'QUESTIONED',
}

export class ExecuteRequest {
  ids: string[] = [];
}

File: backend/domain/checklist/checklist.go
// SPDX-FileCopyrightText: 2025 Mercedes-Benz Group AG and Mercedes-Benz AG
//
// SPDX-License-Identifier: Apache-2.0

package checklist

import (
	"mercedes-benz.ghe.com/foss/disuko/domain"
	"mercedes-benz.ghe.com/foss/disuko/domain/project"
)

type TriggerType string

const (
	Default           TriggerType = "DEFAULT"
	ClassificationAND TriggerType = "CLASS_AND"
	ClassificationOR  TriggerType = "CLASS_OR"
	PolicyStatus      TriggerType = "POLICY_STATUS"
	ScanRemark        TriggerType = "SCAN_REMARK"
	License           TriggerType = "LICENSE"
	ComponentName     TriggerType = "COMPONENT_NAME"
)

type PolicyStatusType string

const (
	Allowed    PolicyStatusType = "ALLOWED"
	Denied     PolicyStatusType = "DENIED"
	Unasserted PolicyStatusType = "UNASSERTED"
	Warned     PolicyStatusType = "WARNED"
	Questioned PolicyStatusType = "QUESTIONED"
)

type Item struct {
	domain.ChildEntity `bson:"inline"`
	Name               string
	TriggerType        TriggerType
	Classifications    []string
	PolicyStatus       []PolicyStatusType
	PolicyLabels       []string
	ScanRemarks        *project.ScanRemarkStatus
	LicenseIds         []string
	ComponentNames     []string
	TargetTemplateName string
	TargetTemplateKey  string
}

type Checklist struct {
	domain.RootEntity `bson:"inline"`
	Name              string
	NameDE            string
	Description       string
	DescriptionDE     string
	PolicyLabels      []string
	Items             []Item
	Active            bool
}
